src/libponyc/pass/expr.c

```diff
@@ -139,6 +139,7 @@ bool is_method_result(typecheck_t* t, ast_t* ast)
     case TK_IF:
     case TK_WHILE:
     case TK_MATCH:
+    case TK_IFDEF:
```

Comment:
Are there any other cases known missing from this function?

Reply:
Referencing `treecheckdef.h`, here are things I'm not seeing in this function that probably should be here:
- `TK_FOR` - the first two children are part of iterator declaration, and should return false here
- `TK_WITH` - the first child should return false
- `TK_REPEAT` - the second child (the condition) should return false